### ISSUE TITLE
Ever consider that user just want to keep an older version of a package?

### ISSUE THREAD

**Issue by caracare521** [EvolveLabs/electron-updater#23]
This is the satisfy function in check.js 
```
function satisfy(registry, name, desired, current, exists, callback) {
  var url = registry + '/' + name
  got(url, {json:true}, function (err, data) {
    if(err) return callback(err)
    var versions = data.versions
    var available;
    Object.getOwnPropertyNames(data.versions).forEach(function (v) {
      if(semver.satisfies(v, desired) && (!available || semver.gt(v, available))) {
        available = v
      }
    })
  
    if(available && (!exists || semver.gt(available, current))) {      
      return callback(null, {
        name: name,
        desired: desired,
        current: current,
        available: available
      })
    }

    // We already have the most up-to-date version
    callback()
  })
}
```

when there is an greater available version, it will tell the app to update as there is a dependency package need to be updated.

Here is what I captured in my log.
![image](https://cloud.githubusercontent.com/assets/7552623/9751630/a1ff9e80-56d9-11e5-991c-4f93af8a6985.png)

My current equals my desired, but  smaller than available. This will trigger an update.

Looks like "var desired" is not used? Sometimes user may just want to keep a stable version for some pkgs for a stable app.

**Comment by justinmchase** [comment]
The way it's working right now is that it puts the power of when and what to update in your hands by allowing you to use semantic versioning in your package.json file. Specifically it's using [semver](https://www.npmjs.com/package/semver) to resolve the version. In your case you're asking for version `~2.1.1` which is a [tilde range](https://www.npmjs.com/package/semver):

> Tilde Ranges ~1.2.3 ~1.2 ~1
> Allows patch-level changes if a minor version is specified on the comparator. Allows minor-level changes if not.

Or in your case:
```
~2.1.1 := >=2.1.1 <2.2.0
```

Which means when a patch level change comes out (e.g. 2.1.1) it detects that as a newer version for updating and therefore updates. This is not a bad thing, as patch level changes are supposed to be backwards compatible fixes only. However it does trigger a "full update" when 3rd party packages  are updated, which you may not want to have happen.

If you don't want to update on patch level changes you can change the version to be simply `2.1.1` (no tilde). And then when a patch level release is available for that package it won't detect it as a valid version for that range. You would manually update to newer versions at a time of your choosing and publish an update to your app or plugin.

Something like this:
```
$ npm outdated
Package     Current  Wanted  Latest  Location
jquery          2.1.1   2.1.1   2.1.2  jquery
$ npm install jquery@2.1.2 --save
```

That will install the new version of jquery and update your package.json. Next time you publish an update to your app / plugin, it will also pull down the new version of jquery, but not sooner.

Does this sound like a good way to manage dependency updates? Or do you still think that dependencies should not trigger an update available assuming the version that's there satisfies the desired version?

Basically we would just have to remove:
```
semver.gt(v, available)
```

From the satisfy function for dependencies. You still want it for the app / plugins themselves, but not their dependencies. What do you think?

**Comment by caracare521** [comment]
What I was thinking is once your local pkg version satisfies the range listed in package.json, then it should not be updated. I just add an if statement to check before got(url, {json:true}, function (err, data)  from the satisfy function in check.js. If current pkg passes this satisfaction check, it will do a callback() directly which will not add anything  to the dependencies array. So it won't trigger an update. 

This method is just like what you mentioned above. Just remove the "~" to make it  just simple like 2.1.1 in package.json. 

After I read what you wrote in your reply, I feel that maybe this should be controlled by users depending on their own needs. 

I agree  usually patch level changes are  backwards compatible. And keep it up-to-date is good for research and development. 

But when it comes to industry, too many updates may buggy app users.
At the same time, app owners need to pay for these updates if they use a paid service provided by CDN vendors.

**Comment by justinmchase** [comment]
I think I'm going to close this one as won't fix actually. If you don't want to trigger updates on patch level changes for dependencies then you can control this by locking down your dependencies to specific versions (e.g. remove `~` and `^`, etc.)

**Comment by caracare521** [comment]
Yeah we can do that.
